src/pveclib/vec_f128_ppc.h

```diff
@@ -951,17 +1314,352 @@ test_cmpltf128_v3d (vui128_t vfa128, vui128_t vfb128)
+ *
+ * Convertions between quad-precision and quadword integers is
+ * complicated by the fact that the QP significand is only 113-bits
+ * while the quadword integer magnitude can be 127/128 bits.
+ * It may not be possible to represent the quadword magnitude
+ * exactly.
+ * Conversions from quad-precision float to integer may have
+ * nonzero fractions which require rounding/truncation.
+ *
+ * For POWER9 we have the
+ * <B>VXS Scalar Convert with round to zero Quad-Precision to
+ * Signed/Unsigned Doubleword
+ * <I>(xscvqpsdz/xscvqpudz)</I></B> instructions.
+ * For POWER10 we have the
+ * <B>VXS Scalar Convert with round to zero Quad-Precision to
+ * Signed/Unsigned Quadword
+ * <I>(xscvqpsqz/xscvqpuqz)</I></B> instructions.
+ * Conversion using other rounding modes require using
+ * <B>VSX Scalar Round to Quad-Precision Integer <I>(xsrqpi)</I></B>
+ * instruction.
+ *
+ * \note The <I>xsrqpi</I> instruction
+ * allows for overriding the rounding mode as an immediate operand.
+ * So a two instruction sequence can implement any of the four
+ * <B>FPSCR<sub>RN</sub></B> rounding modes plus the fifth
+ * (Round to Nearest Away) mode specific to floating point integer
+ * instructions.
  *
- * \subsubsection f128_softfloat_0_0_2_2 Convert Quadword integer to Quad-Precision
+ * For this example we will look at Convert with round to zero
+ * Quad-Precision to Unsigned Quadword.
+ * The POWER10 operation can be implemented as a single xscvqpuqz
+ * instruction. For example:
+ * \code
+static inline vui128_t
+vec_xscvqpuqz (__binary128 f128)
+{
+  vui128_t result;
+#if defined (_ARCH_PWR10)  && (__GNUC__ >= 10)
+  __asm__(
+      "xscvqpuqz %0,%1"
+      : "=v" (result)
+      : "v" (f128)
+      : );
+#else
+...
+#endif
+  return result;
+}
+ * \endcode
+ * We use in-line assembler here as there are no current or planed
+ * compiler intrinsics for this and the C language only supports
+ * conversions between __float128 and __int128 scalars.
+ * The scalar conversions returns the __int128 result in GPR pair,
+ * while we need the result in vector register.
+ *
+ * \note We could try to implement the POWER9 convert to quadword
+ * operation using two xscvqpudz instructions (at 12-cycles each).
+ * But this also requires two QP-multiplies (at 24-cycles each),
+ * plus xscvudqp/xssubqp (at 12-cycles each).
+ * So far it looks like using the POWER8 implementation for POWER9
+ * will actually perform better.
+ *
+ * The POWER8 implementation looks like this:
+ * \code
+  vui64_t q_exp, q_delta, x_exp;
+  vui128_t q_sig;
+  vb128_t b_sign;
+  const vui128_t q_zero = { 0 };
+  const vui128_t q_ones = (vui128_t) vec_splat_s32 (-1);
+  const vui64_t exp_low = (vui64_t) CONST_VINT64_DW( 0x3fff, 0x3fff );
+  const vui64_t exp_high = (vui64_t) CONST_VINT64_DW( (0x3fff+128), (0x3fff+128) );
+  const vui64_t exp_127 = (vui64_t) CONST_VINT64_DW( (0x3fff+127), (0x3fff+127) );
+  const vui64_t q_naninf = (vui64_t) CONST_VINT64_DW( 0x7fff, 0x7fff );
```

Comment:
nit: you've used &lt;space&gt;&lt;paren&gt; almost uniformly, but here you use &lt;paren&gt;&lt;space&gt;

Reply:
Ok But these these changes ripple everywhere 

Much bigger patch now.